desktop_version/src/Screen.cpp

```diff
@@ -109,6 +89,31 @@ void Screen::init(
 	ResizeScreen(windowWidth, windowHeight);
 }
 
+void Screen::LoadIcon()
+{
+	unsigned char *fileIn = NULL;
+	size_t length = 0;
+	unsigned char *data;
+	unsigned int width, height;
+	FILESYSTEM_loadFileToMemory("VVVVVV.png", &fileIn, &length);
+	lodepng_decode24(&data, &width, &height, fileIn, length);
+	FILESYSTEM_freeMemory(&fileIn);
+	SDL_Surface *icon = SDL_CreateRGBSurfaceFrom(
+		data,
+		width,
+		height,
+		24,
+		width * 3,
+		0x000000FF,
+		0x0000FF00,
+		0x00FF0000,
+		0x00000000
+	);
+	SDL_SetWindowIcon(m_window, icon);
+	SDL_FreeSurface(icon);
+	SDL_free(data);
```

Comment:
In this one case it does have to be free - we would need to macro the malloc call in lodepng to get this working with SDL_stdlib.

Reply:
Fixed, rebased, and forced-pushed.